File: src/System/InstructorRegistrationSystem.java
package System;

import java.io.FileNotFoundException;
import java.util.Optional;
import java.util.Scanner;

import Enums.FilterType;
import data.DataManager;
import person.Instructor;

//Kaan Camci 150119063
public class InstructorRegistrationSystem {

	private Scanner scanner;
	private RegistrationSystem registrationSystem1;

	public InstructorRegistrationSystem(RegistrationSystem registrationSystem) throws FileNotFoundException {
		scanner = new Scanner(System.in);
		registrationSystem1 = registrationSystem;
		instructorLogin();
	}

	private void instructorLogin() throws FileNotFoundException {
		Instructor currentUser = null;

		System.out.println("Please provide your ID:");
		System.out.println("----\nSuggestion: Enter \"????\"");
		while (true) {
			String providedID = scanner.nextLine();
			Optional<Instructor> currentOptionalAdvisor = DataManager.getInstance().findInstructor(providedID,
					FilterType.ID);
			if (currentOptionalAdvisor.isPresent()) {
				currentUser = currentOptionalAdvisor.get();
				instructorMenu(currentUser);
				break;
			} else {
				System.out.print("Instructor not found, please try again: ");
			}
		}
	}

	private void instructorMenu(Instructor currentUser) throws FileNotFoundException {

		boolean validInput = false;

		int menuChoice = 0;

		while (menuChoice != 3) {

			System.out.println("Please choose a menu: ");
			System.out.println("1-Show Lecture Sessions");
			System.out.println("2-Show Students Of A Lecture Session");
			System.out.println("3-Sign Out");

			validInput = false;

			while (!validInput) {

				menuChoice = scanner.nextInt();

				switch (menuChoice) {
				case 1:
					validInput = true;
					currentUser.showLectures();
					break;
				case 2:
					validInput = true;
					currentUser.showStudents();
					break;
				case 3:
					validInput = true;
					signOut();
					break;
				default:
					System.out.print("The input is not valid, please provide a valid input.");
				}
			}
		}

	}

	private void signOut() throws FileNotFoundException {
		registrationSystem1.menu();
	}

}

File: pythonCode/InstructorRegistrationSystem.py
from FilterType import FilterType
from DataManager import DataManager
from Logger import Logger

class InstructorRegistrationSystem():

    #Holding registrationSystem object for returning after signing out

    def __init__(self, registrationSystem):
        #Constructor of InstructorRegistrationSystem

        self.__log = Logger.getLogger("logs")
        self.__registrationSystem1 = registrationSystem

    def instructorLogin(self):

        #Searching for a Instructor with a given id. If there is an id equal with providedId currentUser is that Instructor.
        self.__log.info("Please provide an Instructor ID:")
        self.__log.info("----\nSuggestion: Enter \"150097\"")
        while True:
            providedID = input()
            currentUser = DataManager.getInstance().findAdvisor(providedID, FilterType.ID)
            self.instructorMenu(currentUser)

    def instructorMenu(self, currentUser):
        #Showing Instructor's menu choices and making a choice
        validInput = False
        menuChoice = str(0)
        while menuChoice != str(3):
            self.__log.info("Choose a menu: ")
            self.__log.info("1-Show Lecture Sessions")
            self.__log.info("2-Show Students Of A Lecture Session")
            self.__log.info("3-Sign Out")

            validInput = False
            while not validInput:
                menuChoice = input()
                if menuChoice==str(1):
                    validInput = True
                    self.showLectures(currentUser)
                elif menuChoice==str(2):
                    validInput = True
                    self.showStudents(currentUser)
                elif menuChoice==str(3):
                    validInput = True
                    self.signOut()
                else:
                    self.__log.info("The input is not valid, please provide a valid input.")

    def showLectures(self, currentUser):
        # generated source for method showLectures 
        schedule = currentUser.getSchedule()
        listOfLectureSessions = schedule.getListOfLectureSessions()
        for lectureSession in listOfLectureSessions:
            self.__log.info(lectureSession.getLecture().getName() + "." + lectureSession.getSessionID())
            self.__log.info("---------------------------------------")
            self.__log.info("Quota: " + lectureSession.getLecture().getQuota())
            self.__log.info("Number Of Students: " + lectureSession.getListOfStudents().size() + "\n")
            self.__log.info("---------------------------------------")

        schedule.showSchedule()

    def showStudents(self, currentUser):
        # generated source for method showStudents 
        count = 0
        for lectureSession in currentUser.getSchedule().getListOfLectureSessions():
            count += 1
            self.__log.info(count + ". " + lectureSession.getLecture().getName() + "." + lectureSession.getSessionID())
        self.__log.info("Choose A Session: ")
        sessionChoice = input()
        lectureSession = currentUser.getSchedule().getListOfLectureSessions().get(sessionChoice - 1)
        for student in lectureSession.getListOfStudents():
            self.__log.info("ID: " + student.getID() + "Name: " + student.getFullName())


    def signOut(self):
        #returning back to RegistrationSystem's menu
        self.__registrationSystem1.menu()
